### ISSUE TITLE
Unable to install octave on OSX 10.11.6

### ISSUE THREAD

**Issue by DomT4** [Homebrew/homebrew-science#4075]
_From @jackson312 on August 19, 2016 0:2_

If Homebrew was updated on Aug 10-11th 2016 and `brew update` always says `Already up-to-date.` you need to run: `cd $(brew --repo); git fetch; git reset --hard origin/master; brew update`.

# Please follow the general troubleshooting steps first:

- [x ] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?

### Bug reports:

I am unable to install octave on OSX 10.11.6.

I have tried `brew uninstall -f $(brew deps -r octave)` but this did not help
https://gist.github.com/c25181835983b286ac686566dfa16ab8

_Copied from original issue: Homebrew/homebrew-core#4030_

**Comment by dpo** [comment]
@jackson312 your issue should be fixed by #4071.

**Comment by jackson312** [comment]
I am still seeing the error, when will this get merged into the main repo?

**Comment by schoeps** [comment]
Should be ready very soon. In super urgent cases you can do stuff like `brew install https://raw.githubusercontent.com/schoeps/homebrew-science/2f1b48ef7797acad1cdce51565713b5610e5957d/octave.rb --HEAD --with-gui`

**Comment by dpo** [comment]
@jackson312 Please `brew update` and try to reinstall.

**Comment by muammar** [comment]
I have just tried, but there is still an error:

```
muammar@zarathustra ~
  % brew -v install octave --with-gui                                                                                                                  !11577
Error: Homebrew dropped support of Qt4 but Octave 4.2 is not yet released.
       If you need the GUI, you may try installing with --HEAD
Error: Kernel.exit
```

**Comment by schoeps** [comment]
I have rarely seen error messages so instructive: "you may try with --HEAD". So please try
`brew -v install octave --with-gui  --HEAD`
or go without GUI
`brew -v install octave`

**Comment by muammar** [comment]
I uninstalled dependencies:

```
brew uninstall -f $(brew deps -r octave)
```

Then, I tried with the command you proposed:

```
brew install https://raw.githubusercontent.com/schoeps/homebrew-science/2f1b48ef7797acad1cdce51565713b5610e5957d/octave.rb --HEAD --with-gui
```

I got this error:

```
Last 15 lines from /Users/muammar/Library/Logs/Homebrew/octave/03.make:
  CXX      src/src_octave_cli-main-cli.o
  CXX      src/src_octave_cli-octave-build-info.o
  GEN      src/octave-config.cc
  GEN      etc/icons/octave-logo-256.png
./build-aux/missing: line 81: rsvg-convert: command not found
WARNING: 'rsvg-convert' is missing on your system.
         You might have modified some files without having the proper
         tools for further handling them.  Check the 'README' file, it
         often tells you about the needed prerequisites for installing
         this package.  You may also peek at any GNU archive site, in
         case some other package contains this missing 'rsvg-convert' program.
make[2]: *** [etc/icons/octave-logo-256.png] Error 127
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

READ THIS: https://git.io/brew-troubleshooting
```

Then I installed: `brew install librsvg`, and retried. Now it works.

<img width="1280" alt="screen shot 2016-08-22 at 18 50 34" src="https://cloud.githubusercontent.com/assets/954209/17863288/5f17e2ee-6899-11e6-92f1-3d19bb3e4ebc.png">

**Comment by schoeps** [comment]
great 👍  However, you do not to reference the explicit `http://..octave.rb` anymore since the new `octave.rb` is now merged into homebrew/science, so  `brew -v install octave --with-gui --HEAD` gives the same result.

Thanks for the remark on `librsvg`. I will add it as dependency for `HEAD`. Maybe you want to become a homebrew contributor and do it by yourself? I would recommend to add `depends_on "librsvg"      => :build` in line 55 of `octave.rb`. The general procedure is described in https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md (you may have to adapt it for `homebrew/science`)

**Comment by muammar** [comment]
> great 👍 However, you do not to reference the explicit http://..octave.rb anymore since the new octave.rb is now merged into homebrew/science, so brew -v install octave --with-gui --HEAD gives the same result.
> 

Thanks for the information about it. 

> Thanks for the remark on librsvg. I will add it as dependency for HEAD. Maybe you want to become a homebrew contributor and do it by yourself? I would recommend to add depends_on "librsvg" => :build in line 55 of octave.rb. The general procedure is described in https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md (you may have to adapt it for homebrew/science)
 
That would be nice. Let me try it if you don't mind :). Thanks!.

**Comment by muammar** [comment]
@schoeps I have created a [PR](https://github.com/Homebrew/homebrew-science/pull/4087).

**Comment by dpo** [comment]
Thank you!

**Comment by jackson312** [comment]
I am getting a timeout. Maybe the fact that I am building using --with-openblas is causing it?

jackson@ingma:~$ brew -v install octave --with-openblas --with-gui --HEAD
==> Installing octave from homebrew/science
==> Using Homebrew-provided fortran compiler.
This may be changed by setting the FC environment variable.
==> Cloning http://www.octave.org/hg/octave
/usr/local/opt/mercurial/bin/hg clone http://www.octave.org/hg/octave /Users/jackson/Library/Caches/Homebrew/octave--hg
abort: error: Operation timed out
Error: Failed to download resource "octave"
Failure while executing: /usr/local/opt/mercurial/bin/hg clone http://www.octave.org/hg/octave /Users/jackson/Library/Caches/Homebrew/octave--hg
jackson@ingma:~$ brew install https://raw.githubusercontent.com/schoeps/homebrew-science/2f1b48ef7797acad1cdce51565713b5610e5957d/octave.rb --HEAD --with-gui --with-openblas

**Comment by dpo** [comment]
In a terminal, are you able to execute the following
```
hg clone http://www.octave.org/hg/octave /tmp/octave--hg
```

**Comment by jackson312** [comment]
No, it times out as well. Maybe they have some ports blocked on the firewall here at work. I can try it tonight at home.

**Comment by dpo** [comment]
At any rate, the problem isn't related to Homebrew. Perhaps the octave tracker will have more information?!

**Comment by muammar** [comment]
I also think is a network problem. In my case, hg is cloning ok.

**Comment by muammar** [comment]
It is not building anymore: 

```
==> Using Homebrew-provided fortran compiler.
This may be changed by setting the FC environment variable.
==> Cloning http://www.octave.org/hg/octave
Updating /Users/muammar/Library/Caches/Homebrew/octave--hg
==> Checking out branch default
==> Downloading https://savannah.gnu.org/bugs/download.php?file_id=38206
Already downloaded: /Users/muammar/Library/Caches/Homebrew/octave--patch-8307cec2b84fe546c8f490329b488ecf1da628ce823301b6765ffa7e6e292eed.php
==> Patching
==> Applying download.php
patching file libgui/src/octave-dock-widget.cc
==> ./bootstrap
==> ./configure --prefix=/usr/local/Cellar/octave/HEAD-6cbbd550aa91_1 --enable-dependency-tracking --enable-link-all-dependencies --enable-shared --disable-st
==> make all
Last 15 lines from /Users/muammar/Library/Logs/Homebrew/octave/03.make:
./liboctave/util/str-vec.h:131:16: note: candidate template ignored: substitution failure [with String_Container = set]: too few template arguments for class template 'set'
string_vector::string_vector (const String_Container<std::string>& lst)
               ^                    ~~~~~~~~~~~~~~~~
./liboctave/util/str-vec.h:40:3: note: candidate constructor not viable: requires 0 arguments, but 1 was provided
  string_vector (void) : Array<std::string> () { }
  ^
./liboctave/util/str-vec.h:67:3: note: candidate constructor not viable: requires 2 arguments, but 1 was provided
  string_vector (const char * const *s, octave_idx_type n);
  ^
  CXX      libinterp/octave-value/libinterp_octave_value_liboctave_value_la-ov-complex.lo
1 error generated.
make[2]: *** [libinterp/octave-value/libinterp_octave_value_liboctave_value_la-ov-classdef.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

READ THIS: https://git.io/brew-troubleshooting
```

**Comment by schoeps** [comment]
This is an octave issue (and I thibk it is already reported upstream). Keep in mind that you are building a development snapshot if using --HEAD.
